docs/source/usage/security_recommendations.rst

```diff
@@ -0,0 +1,113 @@
+Security Recommendations
+========================
+
+BIOS Settings Reset
+-------------------
+
+A lessee leasing a node can change the BIOS settings. If these settings are not reset after the node is returned, then this can result in unexpected behavior for future users of the node. Fortunately Ironic allows nodes to `specify BIOS settings`_ that are applied upon cleaning.
+This feature has been tested and the details are listed below.
+
+Prerequisites
+~~~~~~~~~~~~~
+Enable a specific hardware type that supports BIOS configuration and then enable bios interface. `iDrac`_ hardware type is used here.
+
+* Install python-dracclient:
+
+.. prompt:: bash $
+
+  sudo pip install 'python-dracclient>=3.1.0'
+
+* Enable the idrac hardware type with the required interfaces in ironic.conf:
+
+.. prompt::
+
+  [DEFAULT]
+  enabled_hardware_types=idrac
+  enabled_management_interfaces=idrac-wsman
+  enabled_power_interfaces=idrac-wsman
+  enabled_vendor_interface=idrac-wsman
+  enabled_bios_interface=idrac-wsman
+
+* Restart the ironic conductor service for the new configuration to take effect.
+
+Set the Node
+~~~~~~~~~~~~
+The following command sets a baremetal node with the idrac hardware type using WSMAN for all interfaces:
+
+.. prompt:: bash $
+
+  openstack baremetal node set \
+    --driver idrac \
+    --power-interface idrac-wsman \
+    --management-interface idrac-wsman \
+    --vendor-interface idrac-wsman \
+    --console-interface no-console \
+    --bios-interface idrac-wsman \
+    --driver-info drac_username=<drac_user> \
+    --driver-info drac_password=>drac_passwd> \
+    --driver-info drac_address=<drac_ip> \
+    <node>
+
+Cleanup BIOS Settings
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+After the node is returned, owner can retrieve the cached BIOS configuration:
+
+.. prompt:: bash $
+
+  openstack baremetal node bios setting list (-f json) <node>
+
+Note: node's owner/lessee can retrieve BIOS settings with the default ironic policy rule:
+
+.. prompt::
+
+  #"baremetal:node:bios:get": "rule:is_admin or rule:is_observer"
```

Comment:
You may not need to mention this rule; however, if there's a "baremetal:node:bios:set" I'd mention that under "Prerequisites" along with the optional step of modifying the rule to allow an owner to set the bios settings.

Reply:
Got it. There's no baremetal:node:bios:set policy so I'll remove this.

Reply:
Oh, maybe you should leave this in then. I may have misunderstood - can an owner set the cleaning step to apply a specific bios configuration for only their node?

Reply:
Yes. The manual cleaning step "apply configuration" can set specific bios configurations.

Reply:
Is it possible for a node owner to set this step to automatically run for only their node with their specific configuration?